### ISSUE TITLE
[bug] Cast isnt listed on phone after TV goes idle or app is exited

### ISSUE THREAD

**Issue by AlexeSimon** [yuliskov/SmartTubeNext#1066]
Hello,

I have an issue with **casting** to my Sony BRAVIA while using **SmartTubeNext**.

Issue : The option to cast to the app disappears from the phone Youtube / Vanced app after some time, typically after jumping to another app or turning off the tv (not a real turn off, it puts the tv on a "sleep like" mode).

Solutions : Force stopping the app to restart it or doing a full restart of the tv solves the issue momentarily. It shows again the option on the phone. 

Tested : I tested the setting cast > run in background both on and off, I tried to remove all device and link again, I tried with two different phones and I tried with vanilla Youtube and Youtube Vanced (root), all without success.

Request : please investigate & fix or advise.

**Comment by elpreto** [comment]
I also hace this same Issue in a TV with Android 9.
I just link the phone every time I open the app, but is obviously not very comfortable.

**Comment by Mister-XY** [comment]
Same here with fire tv cube.

**Comment by Omp26** [comment]
I'm having the same issue too.

**Comment by 1Cr** [comment]
Same with my Android TV. Also, once the device gets unlinked / mysteriously disappears, it's impossible to link it again with a code. After entering the code, it tries to connect but never succeeds.

**Comment by AlexeSimon** [comment]
Try force stopping then restarting the app

**Comment by 1Cr** [comment]
> 

Force stopping and restarting doesn't change anything, unfortunately.

**Comment by AlexeSimon** [comment]
Of course! I am just giving advices as your situation seem worse than mine.

I meant restarting the app /on the tv/. Find how to access the app list and force stop it, then start it again, and it should link all right with Vanced. This is what I do now, and I stopped fully restarting the TV (which takes longer).

**Comment by Mister-XY** [comment]
You don't have to restart the app every time. It is sufficient if you briefly deactivate the remote control under Settings -> Remote control and reactivate it.

**Comment by GitStudying** [comment]
Your issue seems a duplicate of https://github.com/yuliskov/SmartTubeNext/issues/1280, to reiterate:
I am quite sure this is not a issue with SmartTubeNext but just with configuration of your chromecast (with Google tv)
I was looking for this a while ago but finally have the proper configuration to make it really convenient. SmartTubeNext should have the display over other apps permission (settings>apps>special apps access>display over other apps>smarttubenext to on). Then go to turn screen on (settings>apps>special apps access>turn screen on) and also toggle STN to on. Then you can cast from screen off or screen on on any screen and cast works like it would on normal youtube. I also installed ATV STN bridge and uninstalled Youtube, I am not sure if that is necessary or not but if it does not work with the above settings you should do that. If you have any questions I can help you more.

**Comment by Omp26** [comment]
> Your issue seems a duplicate of #1280, to reiterate: I am quite sure this is not a issue with SmartTubeNext but just with configuration of your chromecast (with Google tv) I was looking for this a while ago but finally have the proper configuration to make it really convenient. SmartTubeNext should have the display over other apps permission (settings>apps>special apps access>display over other apps>smarttubenext to on). Then go to turn screen on (settings>apps>special apps access>turn screen on) and also toggle STN to on. Then you can cast from screen off or screen on on any screen and cast works like it would on normal youtube. I also installed ATV STN bridge and uninstalled Youtube, I am not sure if that is necessary or not but if it does not work with the above settings you should do that. If you have any questions I can help you more.

Thanks for the help! The STN cast to TV cast link seems to be appearing once the TV is on, so still need to figure out how to cast when the screen's off. I was able to activate "Display over other apps" for STN, but don't see the setting for "Turn Screen On" on my TV (Sony X750H). Do you have a link for ATV STN Bridge? Maybe that could work.
